cl/citations/api_views.py

```diff
@@ -0,0 +1,209 @@
+    pagination_class = MediumAdjustablePagination
+    serializer = OpinionClusterSerializer
+
+    def get(self, request: Request, *args, **kwargs) -> HttpResponse:
+        query = request.query_params
+
+        # Uses the serializer to perform object level validations
+        citation_serializer = CitationRequestSerializer(data=query)
+        citation_serializer.is_valid(raise_exception=True)
+
+        # Get query parameters from the validated data
+        data = citation_serializer.validated_data
+        self.full_text_citation = data.get("text_citation", None)
+        self.reporter = data.get("reporter", None)
+        self.volume = data.get("volume", None)
+        self.page = data.get("citation_page", None)
+
+        if self.full_text_citation:
+            citations = eyecite.get_citations(self.full_text_citation)
+            if not citations:
+                raise NotFound(
+                    f"Unable to find a citation withing the provided text"
+                )
+
+            if citations[0].groups:
+                c = citations[0]
+                self.reporter = slugify(c.groups["reporter"])
+                self.volume = c.groups["volume"]
+                self.page = c.groups["page"]
+            else:
+                raise NotFound(
+                    f"The provided text is not a valid citation. Please review it and try again"
+                )
+
+        self.reporter_slug = slugify(self.reporter)
+
+        # Look up the reporter to get its proper version (so-2d -> So. 2d)
+        proper_reporter = SLUGIFIED_EDITIONS.get(self.reporter_slug, None)
+        if not proper_reporter:
+            proper_reporter = self._attempt_reporter_variation()
+
+        if proper_reporter and self.volume and self.page:
+            return self._citation_handler(request, proper_reporter)
+        else:
+            return self._reporter_volume_handler(request, proper_reporter)
+
+    def _attempt_reporter_variation(self) -> str:
+        """Try to disambiguate an unknown reporter using the variations dict.
+
+        Raises:
+            NotFound: If the unknown reporter string doesn't match a canonical.
+            MultipleChoices: If the reporter variation maps to more than one
+                            canonical reporter.
+
+        Returns:
+            str: canonical name for the reporter.
+        """
+        potential_canonicals = get_canonicals_from_reporter(self.reporter_slug)
+
+        if len(potential_canonicals) == 0:
+            # Couldn't find it as a variation. Give up.
+            raise NotFound(
+                f"Unable to find Reporter with abbreviation of '{self.reporter}'"
+            )
+
+        elif len(potential_canonicals) > 1:
+            # The reporter variation is ambiguous b/c it can refer to more than
+            # one reporter. Abort with a 300 status.
+            raise MultipleChoices(
+                f"Found more than one possible reporter for '{self.reporter}'"
+            )
```

Comment:
Since we're getting rid of support for reporter/volume dyads, I think we can return serialized results for this, but before we do, we need to talk about what the shape of the response should be, in particular when we make it possible to find more than one citation in a block of text. 

I _think_ we need to get to a place where a response is more like this (very rough idea!):

```
POST: 
    text="In 22 U.S. 44 J. Kagan referenced 334 ASDF 1 and 1 P. 4"
RESPONSE:
[
   {
      "citation": "22 U.S. 44",
      "status": 300,
      "clusters: {...2 clusters...}
    },
    {
      "citation": "334 ASDF 1",
      "status": 404,
      "clusters": {},
    },
    {
      "citation": "1 P. 4",
      "status": 200,
      "clusters": {...1 cluster object...},
    }
]
```

What do you think of that? And then the HTTP status for the API call itself could indicate what it usually does (validation errors, bad URL, etc.)?

Reply:
> Since we're getting rid of support for reporter/volume dyads, I think we can return serialized results for this

do you mean returning all the clusters that match the reporter and its potential canonical? 

> I think we need to get to a place where a response is more like this

I like the idea of including the citation, its status, and any related clusters in the response. This approach offers flexibility for handling multiple lookups. However, I noticed that pagination seems to be missing,  wouldn't pagination be helpful for large result sets?

Reply:
Yes, if we get to the point where we have multiple matches for a citation, we should return them all.

I didn't think about pagination. I wonder if it's really needed. We're not providing the text of the opinions, so these responses should be pretty small. Maybe we can get away with not doing it? 

Another thing, would it be helpful to provide start and finish indexes for the citations that we find? Something like:

```
{
  "citation": "1 P. 22",
  "start_index": 32,
  "end_index": 39,
  ...
}
```

Reply:
> Yes, if we get to the point where we have multiple matches for a citation, we should return them all.

got it! 

> Maybe we can get away with not doing it?

You're right about the data of the opinions. Let's move forward with a non-paginated implementation.

> would it be helpful to provide start and finish indexes for the citations that we find?

I like this idea! I'll start working on this tweak.